modules/common/pod/pod.go

```
@@ -47,3 +47,20 @@ func GetPodListWithLabel(
 
 	return podList, nil
 }
+
+// GetPodFQDNList gets a list of pods matching the labels provided and returns a slice of pod FQDNs.
+func GetPodFQDNList(ctx context.Context, h *helper.Helper, namespace string, labelSelector map[string]string) ([]string, error) {
+	var podSvcNames []string
+	var podList *corev1.PodList
+
+	podList, err := GetPodListWithLabel(ctx, h, namespace, labelSelector)
+	if err != nil {
+		return []string{}, fmt.Errorf("Error getting list of pods: %w", err)
+	}
+
+	for _, pod := range podList.Items {
+		podSvcNames = append(podSvcNames, fmt.Sprintf("%s.%s", pod.Spec.Hostname, pod.Spec.Subdomain))
```

I think we should return an error if Hostname or Subdomain is not set